app/queries/walkshed_query.rb

```diff
@@ -0,0 +1,120 @@
+class WalkshedQuery
```

Comment:
This also appears to be a lazy class and can probably be combined with its calling class.